core/src/main/java/hudson/model/UpdateSite.java

```diff
@@ -1043,6 +1043,17 @@ public PluginWrapper getInstalled() {
             return pm.getPlugin(name);
         }
 
+        @Exported
+        public boolean isCompatible() {
+            return isCompatible(new PluginManager.MetadataCache());
+        }
+
+        public boolean isCompatible(PluginManager.MetadataCache cache) {
```

Comment:
Would `@Restricted(NoExternalUse.class)` this, no reason IMO to expose this to plugins.

If we end up not doing that, needs Javadoc + `@since TODO`.

Reply:
I've restricted this one, since in case some plugins want to call this, it seems they should be calling the version without the "cache" param, as is the pattern for similar methods.